hy/compiler.py

```diff
@@ -121,6 +122,13 @@ def _dec(fn):
     return _dec
 
 
+def builds_if(_type, condition):
+    if condition:
+        return builds(_type)
+    else:
+        return lambda fn: fn
```

Comment:
Thanks for the explanation! I think I should test my suggestions before sending :)